<!doctype html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta name="viewport"
        content="width=device-width, user-scalable=no, initial-scale=1.0, maximum-scale=1.0, minimum-scale=1.0">
  <meta http-equiv="X-UA-Compatible" content="ie=edge">
  <title>Document</title>
</head>
<body>
<h1>AAAAAAAAAAAAAAAAAAA</h1>

<script>
  let F1 = function () {
    return new Promise( (resolve, reject) => {
      resolve('result-1')
    })
  }

  let F2 = function () {
    return new Promise( (resolve, reject) => {
      setTimeout( () => {
        resolve('result-2')
      } )
    } )
  }

  let F3 = function () {
    return new Promise( (resolve, reject) => {
      resolve('result-3')
    } )
  }

///////////////////////////////////////
  F1()
  .then( (msg) => {
    console.log(msg)
    return F2()
  } )
  .then( (msg) => {
    console.log(msg);
    return F3()
  } )
  .then( (msg) => {
    console.log(msg)
  } )
  .catch( (err) => {
    console.log(err)
  } )
  // -------------------------------------------
  function f1(){
    return new Promise( (resolve, reject)=> {
      setTimeout( () => {
      resolve('AAAAAAAAAA')
      }, 2000 )
    } )
  }

 async function f2() {
    console.log('bbb')
    let result = await f1();
    console.log(result)
  }
  f2()

  // async function fetchData() {
  //   try {
  //     const response = await fetch('https://api.example.com/data');
  //     const data = await response.json();
  //     console.log('data', data);
  //   } catch (error) {
  //     console.error('error', error);
  //   }
  // }
  //
  // fetchData();
  console.log('CCCC')
  ////////////////////////////////

  async function fetchData() {
    try {
      console.log('11111')
      const response = await fetch('https://jsonplaceholder.typicode.com/posts/1');
      const data = await response.json();
      console.log('222222', data);
    } catch (error) {
      console.error('errrrr', error);
    }
  }

  fetchData();
  ///////////////////////////////
  function wait(ms) {
    return new Promise(resolve => setTimeout(resolve, ms));
  }

  async function exampleAsyncFunction() {
    console.log('Start');
    await wait(2000); // Pause for 2 seconds
    console.log('After 2 seconds');
  }

  exampleAsyncFunction();


  function test() {
    return new Promise( (resolve, reject) => {
      setTimeout( () => {
       resolve('KKKKKKKKKKKKK')
      },3000 )
    } )
  }

  async function test2() {
    console.log('nnnnnnnnnnnnnn')
    let res1 = await test()
    console.log('res1', res1)
  }
  test2()
  ///////////////////////////////// try catch
  try {
    // Code that might throw an error
    const result = someFunction();
    console.log(result); // This won't be reached if an error occurs
  } catch (error) {
    // Handle the error
    console.error("An error occurred:", error.message);
  }

</script>
</body>
</html>